SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -408,16 +433,61 @@ bool FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig() const
 	return true;
 }
 
-void FSpatialGDKEditorToolbarModule::StartSpatialOSButtonClicked()
+void FSpatialGDKEditorToolbarModule::StartSpatialServiceButtonClicked()
 {
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this]
+	{
+		FDateTime StartTime = FDateTime::Now();
+		ShowTaskStartNotification(TEXT("Starting spatial service..."));
+
+		if (!LocalDeploymentManager->TryStartSpatialService())
+		{
+			ShowFailedNotification(TEXT("Spatial service failed to start"));
+			UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Could not start spatial service."));
+			return;
+		}
+
+		FDateTime EndTime = FDateTime::Now();
+		FTimespan Span = EndTime - StartTime;
+
+		ShowSuccessNotification(TEXT("Spatial service started!"));
+		UE_LOG(LogSpatialGDKEditorToolbar, Log, TEXT("Spatial service started in %f secoonds."), Span.GetTotalSeconds());
+	});
+}
+
+void FSpatialGDKEditorToolbarModule::StopSpatialServiceButtonClicked()
+{
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this]
+	{
+		FDateTime StartTime = FDateTime::Now();
+		ShowTaskStartNotification(TEXT("Stopping Spatial service..."));
+
+		if (!LocalDeploymentManager->TryStopSpatialService())
+		{
+			ShowFailedNotification(TEXT("Spatial service failed to stop"));
+			UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Could not stop spatial service."));
+			return;
+		}
+
+		FDateTime EndTime = FDateTime::Now();
+		FTimespan Span = EndTime - StartTime;
```

Comment:
You take the time in a lot of different places. I feel like there must be an easier way with some kind of scoping object or something.